src/sentry/search/events/builder/discover.py

```diff
@@ -86,6 +87,12 @@ def resolve_field(self, raw_field: str, alias: bool = False) -> Column:
 
         return super().resolve_field(raw_field, alias)
 
+    def resolve_projects(self) -> list[int]:
+        if options.get("sentry.search.events.project.check_event"):
+            return [proj.id for proj in self.params.projects if proj.flags.has_transactions]
```

Comment:
Can't discover also query for errors?

Reply:
Good callout, changed this to treat the condition differently based on the dataset selected
Discover -> either an error or a txn has been sent
Events -> an error has been sent
Txn/IssuePlatform -> a txn has been sent